### PR TITLE
Add license

### PR DESCRIPTION
**Pull request by dehanj**
Adding a license to our repo to make it clear what license we release under.

### CODE REVIEW

**Comment by dehanj** [comment]
According to https://choosealicense.com/non-software/ GPL2 seems to be our best choice, since the rest is GPL2.

**Comment by dehanj** [comment]
I re-read the resource I referenced, and realize I might have misunderstood it. When I read it again, it seems like the recommendation is CC and that we should explicitly point out that the code examples are under GPLv2.

What do you think?

**Comment by SallSim** [comment]
Yes, I think CC for the dev handbook is a good choice.

**Comment by dehanj** [comment]
I now added so we have [CC-BY-SA-4.0](https://choosealicense.com/licenses/cc-by-sa-4.0/) as a license for the documentation, and noted that the code examples are licensed under GPLv2. 

I also added a footer on the rendered web pages, so it is clear when browsing the site as well.

Or do you prefer a different version of CC? I'm not sure.

Note that the link in the footer won't work until this is actually merged.

File: LICENSES/CC-BY-SA-4.0.txt [hunk 1]
```diff
@@ -0,0 +1,427 @@
+Attribution-ShareAlike 4.0 International
+
+=======================================================================
+
+Creative Commons Corporation ("Creative Commons") is not a law firm and
```
Comment (mchack-work) [inline LICENSES/CC-BY-SA-4.0.txt:5]: Should the preamble here really be included in a license file for "Creative Commons Attribution-ShareAlike"? It seems to me that the title and the real license begins further down below?

File: LICENSES/CC-BY-SA-4.0.txt [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+     rights in the material. A licensor may make special requests,
+     such as asking that all changes be marked or described.
+     Although not required by our licenses, you are encouraged to
+     respect those requests where reasonable. More considerations
+     for the public:
+     wiki.creativecommons.org/Considerations_for_licensees
+
+=======================================================================
+
+Creative Commons Attribution-ShareAlike 4.0 International Public
```
Comment (mchack-work) [inline LICENSES/CC-BY-SA-4.0.txt:57]: Doesn't the actual license text start here?

File: LICENSES/gpl-v2.only.txt [hunk 3]
```diff
@@ -0,0 +1,339 @@
+                    GNU GENERAL PUBLIC LICENSE
```
Comment (mchack-work) [inline LICENSES/gpl-v2.only.txt:1]: Will Github get that we use different licenses and correctly mark our repo if we use a LICENSE *directory* like this? It has automatic recognition of a `LICENSE` file, but not a directory, right? What's the best course of action here?

File: README.md [hunk 4]
```diff
@@ -60,3 +60,29 @@ your favourite editor.
 In the front matter (the stuff inside `---` in the beginning of the
 file) you can specify the internal order with `weight: 2` to get it in
 second place in the table of contents.
+
+## License
+
+Unless otherwise noted, the content of the Dev Handbook are licensed
+under CC-BY-SA-4.0.
+
+The code examples of the Dev Handbook are licensed under the terms and
```
Comment (mchack-work) [inline README.md:69]: I know we use GPLv2-only in real source, but is it a good idea to have example code under such a restrictive license? This means anyone copying code from the Dev Handbook will immediately be forced to share their source code, right?

Comment (dehanj) [reply README.md:69]: This seems to be the recommended way for documentation, but if we don't think the example code here ever will be substantial we might as well remove GPLv2 and keep everything CC. That is fine for me.

File: LICENSES/CC-BY-SA-4.0.txt (hunk 1 above)
Comment (dehanj) [reply LICENSES/CC-BY-SA-4.0.txt:5]: You might be right. 

I just followed the instructions here: https://choosealicense.com/licenses/cc-by-sa-4.0/
"How to apply this license", I guess I thought the entire text was the license. Maybe it should go.

**Comment by dehanj** [comment]
> I'm fine with continuing as is right now, just pointing out some things we might need to think about. Perhaps merge right now and then relicense after?

Let's do it right, otherwise we just postpone work. 
I think 5 minutes of discussion will suffice to decide what we should change.

**Pull request merged**